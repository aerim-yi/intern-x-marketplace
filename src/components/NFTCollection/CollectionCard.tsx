import React from 'react';
import Card from '../Card/Card';
import './Collection.css';
import placeholderImg from '../../asset/placehoderImg.jpg'

interface Props {
  url: string,
  name: string
}

const CollectionCard = (props: Props) => {
  return (
    <Card className="collections" data-testid="CollectionCard">
      <img alt=""
        src={props.url}
        data-testid="CollectionCard__Img"
        // Load placeholder on error: https://stackoverflow.com/questions/34097560/react-js-replace-img-src-onerror
        onError={({ currentTarget }) => {
          currentTarget.onerror = null;
          currentTarget.src = placeholderImg;
        }}
      />
      <h5 data-testid="CollectionCard_Name">{props.name}</h5>
    </Card>
  );
}

export default CollectionCard;
