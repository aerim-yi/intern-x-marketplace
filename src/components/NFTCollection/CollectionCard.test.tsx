import { render, fireEvent, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom';
import CollectionCard from './CollectionCard'
import placeholderImg from '../../asset/placeholderImg.jpg'


const collectionCardProps = {
    url: 'http.test.com',
    name: 'collectionName'
}

describe('CollectionCard component', () => {
    test('match snapshot', () => {
        const { container } = render(
            <CollectionCard
                name={collectionCardProps.name}
                url={collectionCardProps.url} />
        );
        expect(container).toMatchSnapshot()
    })

    test('Load placeholder image on error', () => {
        render(
            <CollectionCard
                name={collectionCardProps.name}
                url={collectionCardProps.url} />
        );
        const image = screen.getByRole('img');
        fireEvent.error(image)
        expect(image).toHaveAttribute('src', 'placeholderImg.jpg');
    })
})
