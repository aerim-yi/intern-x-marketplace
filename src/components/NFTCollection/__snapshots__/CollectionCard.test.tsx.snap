// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionCard component match snapshot 1`] = `
<div>
  <div
    class="card collections"
  >
    <div
      class="cardWrapper"
    >
      <img
        alt=""
        data-testid="CollectionCard__Img"
        src="http.test.com"
      />
    </div>
    <h5
      class="cardText"
      data-testid="CollectionCard_Name"
    >
      collectionName
    </h5>
  </div>
</div>
`;
